src/contracts/ProofOfPhysicalAddress.contract.js

```diff
@@ -0,0 +1,106 @@
+import Web3 from 'web3'
+import helpers from './helpers'
+
+const CONFIRM_ADDRESS_EVENT_NAME = 'LogAddressConfirmed'
+
+export default class ProofOfPhysicalAddress {
+  async init({ web3, netId, addresses }) {
+    let web3_10 = new Web3(web3.currentProvider)
+    const { PROOF_OF_PHYSICAL_ADDRESS } = addresses
+
+    const branch = helpers.getBranch(netId)
+    if (branch !== 'core') {
+      throw new Error(`ProofOfPhysicalAddress contract not deployed network "${branch}"`)
+    }
+    let proofOfPhysicalAddressAbi = await helpers.getABI(branch, 'ProofOfPhysicalAddress')
+    this.instance = new web3_10.eth.Contract(proofOfPhysicalAddressAbi, PROOF_OF_PHYSICAL_ADDRESS)
+
+    this.getPhysicalAddressesOfWalletAddressArray = this.getPhysicalAddressesOfWalletAddressArray.bind(this)
+    this.getAllEvents = this.getAllEvents.bind(this)
+    this.getPhysicalAddressesByWalletAddressAndKeccakIdentifierArray = this.getPhysicalAddressesByWalletAddressAndKeccakIdentifierArray.bind(
+      this
+    )
+  }
+
+  /**
+   * Given an array of wallet address, return a promise that resolves to an array of result
+   * sets of confirmed addresses from PoBA contract.
+   * @param {String[]} walletAddressArray
+   * @return {Promise}
+   */
+  async getPhysicalAddressesOfWalletAddressArray(walletAddressArray) {
+    let confirmedAddresses = []
+    try {
+      const confirmAddressEvents = await this.getAllEvents(CONFIRM_ADDRESS_EVENT_NAME)
+      const getConfirmedAddressesPromises = walletAddressArray.map(walletAddress => {
+        let confirmedAddressByWalletAddress = null
+        let keccakIdentifiers = confirmAddressEvents
+          .filter(event => event.returnValues.wallet === walletAddress)
+          .map(event => event.returnValues.keccakIdentifier)
+        if (keccakIdentifiers.length > 0) {
+          confirmedAddressByWalletAddress = this.getPhysicalAddressesByWalletAddressAndKeccakIdentifierArray(
+            walletAddress,
+            keccakIdentifiers
+          ).then(physicalAddresses => {
+            return physicalAddresses.length > 0 ? physicalAddresses : null
+          })
+        }
+        return Promise.resolve(confirmedAddressByWalletAddress)
+      })
+      confirmedAddresses = await Promise.all(getConfirmedAddressesPromises)
+    } catch (e) {
+      console.error(`Error in getConfirmedAddressesOfWalletAddressArray`, e)
```

Comment:
The name of the function `getConfirmedAddressesOfWalletAddressArray` seems to be outdated - I don't see this name in the code.